### PR TITLE
Feature/jcs header component

### LINKED ISSUE #3: Search component
- [ ] Add crude jsx
- [ ] Add crude styling
- [ ] Test

### PR DESCRIPTION
**Pull request by JoshSevy**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Added functionality to the Search functional component that is in the Header component
## Type of change 
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [x] No
- [ ] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
React does render but still need to attach api to project for the search functionality to be tested correctly
# Message for reviewer:
Just a start to adding typescript and react hooks to the project. Had to comment out all files that haven't been started yet due to typescript error, that won't allow render of unused components.
# Checklist:
- [ ] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Search/Search.tsx [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+import React, { useState, useEffect } from "react";
 import './Search.scss'
 
+
+const Search: React.FC = () => {
+  const [query, setQuery] = useState<string>('')
+  const [debouncedQuery, setDebouncedQuery] = useState<string>(query)
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
```
Comment (JoshSevy) [inline src/Search/Search.tsx:10]: This is the timeout function for the search component that won't send a request until user is done typing for 750 ms

Comment (leighlars) [reply src/Search/Search.tsx:10]: I like this a lot, removes a lot of the chaos of instantaneous searching

**Review by leighlars — APPROVED** [review]
This looks great, thank you for allowing us to shadow you!

File: src/Search/Search.tsx [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+      setDebouncedQuery(query)
+    }, 750)
```
Comment (ErinUntermeyer) [inline src/Search/Search.tsx:12]: I like the idea of using setTimeout to limit the number of fetch calls made!

File: src/Search/Search.tsx [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
 import './Search.scss'
 
+
+const Search: React.FC = () => {
+  const [query, setQuery] = useState<string>('')
+  const [debouncedQuery, setDebouncedQuery] = useState<string>(query)
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setDebouncedQuery(query)
```
Comment (ErinUntermeyer) [inline src/Search/Search.tsx:11]: Should we start with the search functionality being a drop-down of all supported cities? We could then add functionality that would locate the nearest supported city and display that as a stretch goal?

**Pull request merged**